iridium-core-server/src/main/java/software/iridium/api/service/AuthorizationService.java

```diff
@@ -445,4 +441,50 @@ public String proxyAuthorizationRequestToProvider(
     // todo: throw exception for provider not found
     return null;
   }
+
+  @Transactional(propagation = Propagation.REQUIRED)
+  public AccessTokenResponse refreshToken(Map<String, String> params) {
+
+    checkArgument(
+        attributeValidator.isNotBlank(
+            params.getOrDefault(AuthorizationCodeFlowConstants.GRANT_TYPE.getValue(), "")),
+        "grant_type must not be blank");
+
+    checkArgument(
+        attributeValidator.isNotBlank(
+            params.getOrDefault(AuthorizationCodeFlowConstants.CLIENT_ID.getValue(), "")),
+        "Client ID must not be blank");
```

Comment:
parameter identifiers should match the parameter name for easier debugging.

e.g. `Client ID` should be `client_id`